src/Idris/Parser.idr

```diff
@@ -642,6 +643,22 @@ mutual
       recFieldCompat : Rule Name
       recFieldCompat = dotIdent <|> (symbol "->" *> name)
 
+      parseFieldName : Bool -> Rule (List String)
+      parseFieldName True  = map fieldName <$> [| name :: many recFieldCompat |]
+      parseFieldName False = map fieldName <$> [| name :: pure [] |]
+
+  recordInstance : FileName -> IndentInfo -> Rule PTerm
+  recordInstance fname indents
+      = do b <- bounds (do keyword "record"
+                           n <- name
+                           symbol "{"
+                           commit
+                           fs <- sepBy1 (symbol ",") (field fname indents False)
```

Comment:
I expect we can have record with no fields. Or only implicit fields that
can be inferred. So this should probably be a `sepBy` rather than `sepBy1`.
I have extended the test case with an example that ought to work.

Reply:
One issue here. 

- `elabInstance` disambiguates constructor names by looking at the fields of records of those constructors (if any). It does not look into the expected type. And it does not employ idris's disambiguation process (which I currently don't know much about)
- And it looks like there is no way (at least which I know of) to distinguish between a record constructor and a data constructor via context lookups.

This means in your added test case (`Unit`) it won't be able to disambiguate the constructor from `MkUnit` declared in the `Prelude`

Do we want `elabInstance` to not throw and postpone name resolution in case it's not able to resolve it with provided fields info ?

Reply:
Type-directed disambiguation of record constructor works at the moment
so I think it should also work for this new syntax.

```idris
namespace R1

  public export
  record R1 where
    constructor MkR

namespace R2

  public export
  record R2 where
    constructor MkR

r1 : R1
r1 = MkR

r2 : R2
r2 = MkR
```